pandas/core/indexes/datetimes.py

```diff
@@ -384,6 +387,17 @@ def _is_comparable_dtype(self, dtype: DtypeObj) -> bool:
         """
         Can we compare values of the given dtype to our own?
         """
+        if isinstance(dtype, ArrowDtype):
+            # GH#62277
+            import pyarrow as pa
+
+            pa_dtype = dtype.pyarrow_dtype
+            if not pa.types.is_timestamp(pa_dtype):
```

Comment:
```suggestion
            if dtype.kind != "M":
```

Would allow you to avoid the pyarrow import